src/client/Microsoft.Identity.Client/ClientApplicationBase.cs

```diff
@@ -145,8 +145,22 @@ public async Task RemoveAsync(IAccount account)
 
         private async Task<IEnumerable<IAccount>> GetAccountsInternalAsync(ApiIds apiId, string homeAccountIdFilter = null)
         {
-            var accountsFromCache = await GetAccountsFromCacheAsync(apiId, homeAccountIdFilter).ConfigureAwait(false);
-            var accountsFromBroker = await GetAccountsFromBrokerAsync(homeAccountIdFilter).ConfigureAwait(false);
+            RequestContext requestContext = CreateRequestContext(Guid.NewGuid());
+
+            var authParameters = new AuthenticationRequestParameters(
+                   ServiceBundle,
+                   UserTokenCacheInternal,
+                   new AcquireTokenCommonParameters() { ApiId = apiId },
+                   requestContext,
+                   homeAccountIdFilter);
+
+            // a simple session consisting of a single call
+            var cacheSessionManager = new CacheSessionManager(
+                UserTokenCacheInternal,
+                authParameters);
+
+            var accountsFromCache = await cacheSessionManager.GetAccountsAsync().ConfigureAwait(false);
```

Comment:
>cacheSessionManager [](start = 42, length = 19)

null check?

Reply:
Good practice for methdos that return lists or the sort is to return empty list instead of null, to avoid all these null checks. I will add them for defensive purposes.